tests/test_automation/nightly_olcf_summit.sh

```diff
@@ -43,24 +43,20 @@ echo --- Hostname --- $(hostname -f)
 echo --- Checkout for branch ${branch} $(date)
 echo
 
-# get or refresh the source checkout
-if [ ! -e ${source_dir} ]; then
-  mkdir -p ${base_dir} && cd ${base_dir}
-  echo --- Cloning QMCPACK git $(date)
-  git clone --depth 1 https://github.com/QMCPACK/qmcpack.git ${source_dir}
-else
-  echo --- Updating local QMCPACK git $(date)
-  cd ${source_dir}
-  git pull
-fi
+# refresh the source checkout
+mkdir -p ${base_dir}
+cd ${base_dir}
+rm -rf ${source_dir}
```

Comment:
A bare `rm -rf` feel dangerous - like a mistake in the script could delete things unintentionally.  Perhaps check that the `source_dir` variable is not empty and that that directory exists before doing the remove.

Reply:
both of these cases: `rm -rf    ` and  `rm -rf does_not_exist` result in a no-op with exit `0`

Reply:
I am happy for alternatives that allow it to start with a clean slate.

Reply:
I guess those cases are covered naturally, then.   I don't have any other ideas for checks.